dref/serializers.py

```diff
@@ -428,6 +457,74 @@ def validate(self, data):
+                )
+            if not sub_total:
+                raise serializers.ValidationError({"sub_total": gettext("Sub-total is required for Imminent DREF")})
+            if sub_total != self.SUB_TOTAL:
+                raise serializers.ValidationError(
+                    {"sub_total": gettext("Sub-total should be equal to %s for Imminent DREF" % self.SUB_TOTAL)}
+                )
+            if is_surge_personnel_deployed and not surge_deployment_cost:
+                raise serializers.ValidationError(
+                    {"surge_deployment_cost": gettext("Surge Deployment is required for Imminent DREF")}
+                )
+            if not indirect_cost:
+                raise serializers.ValidationError({"indirect_cost": gettext("Indirect Cost is required for Imminent DREF")})
+            if not total:
+                raise serializers.ValidationError({"total": gettext("Total is required for Imminent DREF")})
+
+            proposed_budget = sum(action.get("budget", 0) for action in proposed_actions)
+            if proposed_budget != sub_total:
+                raise serializers.ValidationError("Sub-total should be equal to proposed budget")
+
+            if is_surge_personnel_deployed:
+                if surge_deployment_cost != self.SURGE_DEPLOYMENT_COST:
+                    raise serializers.ValidationError(
+                        {
+                            "surge_deployment_cost": gettext(
+                                "Surge Deployment Cost should be equal to %s for Surge Personnel Deployed"
+                                % self.SURGE_DEPLOYMENT_COST
+                            )
+                        }
+                    )
+                if indirect_cost != self.INDIRECT_COST_SURGE:
+                    raise serializers.ValidationError(
+                        {
+                            "indirect_cost": gettext(
+                                "Indirect Cost should be equal to %s for Surge Personnel Deployed" % self.INDIRECT_COST_SURGE
+                            )
+                        }
+                    )
+                expected_total = surge_deployment_cost + indirect_cost + sub_total
+            else:
+                if indirect_cost != self.INDIRECT_COST_NO_SURGE:
+                    raise serializers.ValidationError(
+                        {
+                            "indirect_cost": gettext(
+                                "Indirect Cost should be equal to %s for No Surge Personnel Deployed"
+                                % self.INDIRECT_COST_NO_SURGE
+                            )
+                        }
+                    )
+                expected_total = indirect_cost + sub_total
+
+            if expected_total != total:
+                raise serializers.ValidationError(
+                    {"total": gettext("Total should be equal to sum of Sub-total, Surge Deployment Cost and Indirect Cost")}
+                )
```

Comment:
lets make it clear about `total` variable weather it is `total_cost` or something else. Same for other variable.. like `expected_total`,  etc, Also update the validationError after renaming the respective variable.